### PR TITLE
newer golang patch (1.22.7, 1.23.1) sep2024

### LINKED ISSUE #3738: Dependency update - Golang 1.22.7/1.23.1
### Tracking info

Link to any previous tracking issue:  https://github.com/kubernetes/release/issues/3723

Golang mailing list announcement:  https://groups.google.com/g/golang-announce/c/vM0L-2IDlOU/m/R-gi-VMOBQAJ

SIG Release Slack thread: https://kubernetes.slack.com/archives/CJH2GBF7Y/p1725031141201539

### Work items

- [x] `kube-cross`, `go-runner`, `releng-ci` image updates: https://github.com/kubernetes/release/pull/3758

  
  - [x] `kube-cross` image promotion: https://github.com/kubernetes/k8s.io/pull/7324

  
  - [x] `go-runner` image promotion: https://github.com/kubernetes/k8s.io/pull/7323

  
  - [x] `releng-ci` image promotion: https://github.com/kubernetes/k8s.io/pull/7320

#### After go-runner image promotion

- [x] `distroless-iptables` image update: https://github.com/kubernetes/release/pull/3773

  
  - [x] image promotion:  https://github.com/kubernetes/k8s.io/pull/7335

#### After kube-cross and distroless-iptables image promotions

- [x] kubernetes/kubernetes update (`master`):  https://github.com/kubernetes/kubernetes/pull/127611/files

  Ensure the following have been updated within the PR:

  - [x] `.go-version` file
  - [ ] kube-cross image
  - [x] go-runner image
  - [x] distroless-iptables image
  - [x] publishing bot rules
  - [x] test image

> **Note**
> This update may require an update to go.sum files, for example: https://github.com/kubernetes/kubernetes/pull/118507
> This will require an API Review approval.

#### After kubernetes/kubernetes (master) has been updated

- [ ] `k8s-cloud-builder` and `k8s-ci-builder` image updates: 

- [ ] `kubekins`/`krte` image variants update: 

### Cherry picks

- [ ] Kubernetes 1.y-1: 
- [ ] Kubernetes 1.y-2: 
- [ ] Kubernetes 1.y-3: 

- [ ] publishing bot rule updates for active Golang versions: 

#### After kubernetes/kubernetes (release branches) has been updated

- [ ] `k8s-cloud-builder` and `k8s-ci-builder` image updates: 

- [ ] `kubekins`/`krte` image updates: 

### Follow-up items

- [ ] Ensure the Golang issue template is updated with any new requirements
- [ ] <Any other follow up items>

/assign
cc: @kubernetes/release-engineering

### PR DESCRIPTION
**Pull request by haitch**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
- Build Go 1.22.6 images for currently supported Kubernetes release branches
- Build Go 1.23.1 images for currently supported Kubernetes release branches

#### Which issue(s) this PR fixes:

```release-note
Build Go 1.23.1 and 1.22.7 images
```

### CODE REVIEW

**Comment by zhangguanzhang** [comment]
also need for https://github.com/kubernetes/release/blob/master/images/build/distroless-iptables/Makefile#L24

**Comment by ameukam** [comment]
cc @cpanato

File: images/build/go-runner/Makefile
Comment (xmudrii) [inline images/build/go-runner/Makefile:1]: You also need to update `images/build/go-runner/Makefile`

File: images/k8s-cloud-builder/variants.yaml
Comment (xmudrii) [inline images/k8s-cloud-builder/variants.yaml:1]: This should be updated in a dedicated step, after kubernetes/kubernetes is updated to the latest patch releases. Please drop changes to this file as of now

File: images/releng/k8s-ci-builder/variants.yaml
Comment (xmudrii) [inline images/releng/k8s-ci-builder/variants.yaml:1]: This should be updated in a dedicated step, after kubernetes/kubernetes is updated to the latest patch releases. Please drop changes to this file as of now

**Review by xmudrii — CHANGES_REQUESTED** [review]
You also need to update `dependencies.yaml`.

You can take a look at https://github.com/kubernetes/release/pull/3724 as an inspiration how the initial Go update step looks like.

Thank you for the contribution!

**Comment by cpanato** [comment]
I will review this PR this week, most likely tomorrow

/hold

**Comment by cpanato** [comment]
also please rebase and squash the commits

**Comment by haitch** [comment]
thanks for the review @xmudrii @cpanato , working on the rebase now.

**Comment by haitch** [comment]
need ok-to-test label

**Comment by xmudrii** [comment]
We hit the quota on the kube-cross test
/retest

**Comment by xmudrii** [comment]
/release-note-edit
```release-note
Build Go 1.23.1 and 1.22.7 images
```

**Pull request merged**